New Data Source: `azurerm_function_app_host_keys` (#7902)
terraform-providers/terraform-provider-azurerm: website/docs/d/function_app_host_keys.html.markdown

---
subcategory: 'App Service (Web Apps)'
layout: 'azurerm'
page_title: 'Azure Resource Manager: azurerm_function_app_host_keys'
description: |-
  Gets the Host Keys of an existing Function App.
---

# Data Source: azurerm_function_app_host_keys

Use this data source to fetch the Host Keys of an existing Function App

## Example Usage

```hcl
data "azurerm_function_app_host_keys" "example" {
  name                = "example-function"
  resource_group_name = azurerm_resource_group.example.name
}
```

> ~> **Note:** The unencrypted value of FunctionKeys, MasterKey and SystemKeys will be stored in the raw state as plain-text.

Author: great catch! I will fix this!

---
subcategory: 'App Service (Web Apps)'
layout: 'azurerm'
page_title: 'Azure Resource Manager: azurerm_function_app_host_keys'
description: |-

>   Gets HostKeys of an existing Function App.

Reviewer: ```suggestion
  Gets the Host Keys of an existing Function App.
```

---

# Data Source: azurerm_function_app_host_keys

> Use this data source to fetch the Hostkeys of an existing Function App

Reviewer: ```suggestion
Use this data source to fetch the Host Keys of an existing Function App
```

## Example Usage

```hcl
data "azurerm_function_app_host_keys" "example" {

>   name                = "test-azure-functions"

Reviewer: ```suggestion
  name                = "example-function"
```

  resource_group_name = azurerm_resource_group.example.name
}
```

~> **Note:** The unencrypted value of FunctionKeys, MasterKey and SystemKeys will be stored in the raw state as plain-text.

## Argument Reference

The following arguments are supported:

> - `name` - The name of the Function App resource.

Reviewer: can we make this:

```suggestion
- `name` - The name of the Function App.
```

- `resource_group_name` - The name of the Resource Group where the Function App exists.

## Attributes Reference

The following arguments are supported:

- `function_keys` - A key-value pair of Host level function keys.

> - `system_keys` - A key-value pair of system keys

Reviewer: (as above) can we make these specific fields rather than maps

}
```

> ~> **Note:** All arguments including the secret value will be stored in the raw state as plain-text.
> [Read more about sensitive data in state](/docs/state/sensitive-data.html).asterKey and SystemKeys will be stored in the raw state as plain-text.

Reviewer 2: Can we modify the wording here a bit:

```suggestion
~> **Note:** All arguments including the secret value will be stored in the raw state as plain-text, including `default_function_key` and `master_key`. [Read more about sensitive data in state](/docs/state/sensitive-data.html).
```

Author: ![image](https://user-images.githubusercontent.com/42009880/92221509-e35cb000-ee9d-11ea-9e52-0fcd6c77ecac.png)

but I have no problems with changing it if you still wish so
